scripts/reana_bench.py

```diff
@@ -381,39 +376,78 @@ def _save_original_results(workflow: str, df: pd.DataFrame):
 
 
 def submit(
-    workflow_prefix: str, number_of_workflows: int, file: str, workers: int
+    workflow_prefix: str, workflow_range: (int, int), file: str, workers: int
 ) -> None:
     """Submit multiple workflows, do not start them."""
-    if _workflow_already_exists(workflow_prefix):
-        raise Exception("Found duplicated workflow name(s). Please use unique name.")
-
-    _create_and_upload_workflows(workflow_prefix, number_of_workflows, file, workers)
+    _create_and_upload_workflows(workflow_prefix, workflow_range, file, workers)
     logging.info("Finished creating and uploading workflows.")
 
 
-def start(workflow_name: str, workers: int) -> None:
-    """Start already submitted workflows."""
+def _append_to_existing_submit_results(
+    workflow_name: str, new_submit_results: pd.DataFrame
+) -> pd.DataFrame:
+    """Append new submit results to existing submit results and return them."""
 
-    number_of_workflows = len(_get_workflows(workflow_name))
+    submitted_results_path = _build_submitted_results_path(workflow_name)
 
-    if number_of_workflows == 0:
-        raise Exception("Cannot start. Workflow(s) do not exist.")
+    existing_submit_results = pd.DataFrame()
+
+    if submitted_results_path.exists():
+        logging.info("Loading existing submit results. Appending...")
+        existing_submit_results = pd.read_csv(submitted_results_path)
+
+    return existing_submit_results.append(new_submit_results, ignore_index=True)
```

Comment:
Note that appending from several submissions mixes up the `submit_number` in the third column:

```console
$ less test1802_submitted_results.csv                              
name,submit_date,submit_number
test1802-0,2021-11-02T17:03:02,0
test1802-1,2021-11-02T17:03:02,1
test1802-2,2021-11-02T17:05:53,0
test1802-3,2021-11-02T17:05:53,1
test1802-4,2021-11-02T17:05:53,2
test1802-10,2021-11-02T17:06:50,0
test1802-11,2021-11-02T17:06:50,1
```

I guess we don't have to keep it?  The script should be able to deduce "submit numbers" from the `<workflow-name>-<NUMBER>` workflow naming convention already, so keeping `name` and `date` in this CSV file might be enough.

Reply:
It is fine if `submit_number` is mixed - this is known behavior. We first order by `submit_date` and after by `submit_number`. This number exists mainly because our time precision is `seconds` and you can start a couple of workflows in a second (if multiple processes are used), so I added the number to be able to know in which order workflows were started during this second.

*suggestion:* I would prefer not to remove it because it will require unnecessary code changes to a working code (in this PR). I can make it clear in a commit message that `submit number` is only valid within the range it was used in.
Another thing, I can create an issue to investigate later if the submit_number is that useful and if it is not remove it.

What would you prefer? Remove the number or keep it as it is for now?

Reply:
There is no need to create an issue, we can remove it in a later PR that will be touching `collect` `analyze`  (i.e. adding `-n` ranges there), if you prefer doing it that way.  I think `name` and `submit_date` are all we need to identify when a given run N was submitted. There should be no need to store the extra 0,1 that might look artificial that we we won't even use...

Reply:
Okay. Let's do it in later PR. 

In addition, we will need to rename some columns as currently, they are very confusing (like `submit_date`, which in bench CLI means create+upload but in `results` it means "datetime when request to start workflow was sent")

We could have a later PR to harmonize namings + remove `submit_number`.